src/helix.idea.vim

```diff
@@ -0,0 +1,198 @@
+"""""""""""""""""""""""""
+" IdeaVim support       "
+"""""""""""""""""""""""""
+set nohlsearch                    " no highlight search results
+set number                        " line numbers
+set showmode                      " show current mode
+set gdefault                      " substitute all occurrences in line per default
+set history=10000                 " keep x lines of command line history
+set scrolloff=5 sidescrolloff=10  " keep some lines before and after the cursor visible
+set visualbell                    " no beep
+set smartcase                     " no ignore case when pattern is uppercase
+set incsearch                     " show search results while typing
+set wrapscan                      " searches wrap around the end of the file
+
+set ideajoin
+set idearefactormode=keep
+
+" Plugins
+set commentary
+set highlightedyank
+set argtextobj
+set matchit
+set multiple-cursors
+
+nnoremap , <Esc><Esc>
+xnoremap , <Esc><Esc>
+vnoremap , <Esc><Esc>
+
+xnoremap <Esc>o o
+nnoremap o o
+
+nnoremap d vd
+xnoremap d dv
+
+" Movement
+unmap <Left>
+unmap <Right>
+unmap <Down>
+unmap <Up>
+
+noremap t vt
+noremap T vT
+noremap f vf
+noremap F vF
+
+" Select/extend
+nnoremap e ve
+nnoremap E vE
+nnoremap w vw
+nnoremap W vW
+nnoremap b vb
+nnoremap B vB
+
+noremap X 0V
+xnoremap X <nop>
+
+" Match mode
+noremap ma) <Esc>va(
+noremap ma} <Esc>va{
+noremap ma> <Esc>va<
+noremap ma[ <Esc>va[
+
+noremap mi) <Esc>vi)
+noremap mi{ <Esc>vi}
+noremap mi< <Esc>vi>
+noremap mi[ <Esc>vi]
+
+" Goto mode
+nmap gr <Action>(FindUsages)
+nmap gi <Action>(GotoImplementation)
+nmap gd <Action>(GotoDeclaration)
+nmap gT <Action>(GotoTest)
+nmap gp <Action>(PreviousTab)
+nmap gy <Action>(GotoTypeDeclaration)
+
+" Requires plugin: KJump
+nmap gw <Action>(KJumpAction.Word0)
+xmap gw <Action>(KJumpAction.Word0)
+
+" Requires plugin: String manipulation
+xmap & <Action>(StringManipulation.AlignSelections)
+
+" Selection manipulation
+nmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+vmap <A-s> <Action>(EditorAddCaretPerSelectedLine)
+nmap <A-o> <Action>(EditorSelectWord)
+xmap <A-o> <Action>(EditorSelectWord)
+nmap <A-up> <Action>(EditorSelectWord)
+xmap <A-up> <Action>(EditorSelectWord)
+nmap <A-i> <Action>(EditorUnSelectWord)
+xmap <A-i> <Action>(EditorUnSelectWord)
+nmap <A-down> <Action>(EditorUnSelectWord)
+xmap <A-down> <Action>(EditorUnSelectWord)
+nmap C <Action>(EditorCloneCaretBelow)
+xmap C <Action>(EditorCloneCaretAbove)
+
+xnoremap ~ Ugv
```

Comment:
I've noticed that you've introduced several substitutes for commands that are already defined in helix.vim. Do they not work? If so, I would put a comment that explains why it doesn't work (perhaps link to an issue in the emulator or something)

Reply:
Some did not seem to work otherwise, but I think I should go through each of these with an empty config and/or with just `helix.vim` applied. Most of the issues I've ran into were related to `helix.vim` mappings with `<Esc>` and/or mode switching. E.g. the selection disappears just before the command is applied.

Yesterday I fiddled especially with `~` and ``` ` ```. With these changes I get the expected result if the selection is done by an IDE action (e.g. `<Action>(EditorSelectWord)` = "expand selection") and after moving to word start/end with `h`/`l` then selecting the word with `w`/`b`. But if I do the same but explicitly change to visual mode before `w`/`b`, only the first or last letter changes case. So essentially `~`/``` ` ``` first cleared the selection.

Reply:
Actually now that I tried commenting the lines out, I get the exact same behavior. I guess I was trying to fix this issue but only at that point realized the issue with selection. :facepalm:

Reply:
It appears in each of the state that the action fails, prefixing it with `v` fixes the issue. The weird thing is that IdeaVim indicates visual mode is active even before pressing `v`. I think it's in a state of waiting if the previously pressed key is followed by another. There's an indicator in the statusbar that shows what you've typed so far. AFAIK it gets cleared when a mapping gets triggered.

Reply:
They're now removed, but I'll leave this unresolved to remember I need to check the others as well. Thanks!